src/test/java/net/sf/jabref/model/database/BibDatabaseTest.java

```diff
@@ -0,0 +1,124 @@
+package net.sf.jabref.model.database;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.JabRefPreferences;
+import net.sf.jabref.bibtex.BibtexEntryAssert;
+import net.sf.jabref.importer.ParserResult;
+import net.sf.jabref.importer.fileformat.BibtexParser;
+import net.sf.jabref.model.entry.BibEntry;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Collections;
+
+import static org.junit.Assert.*;
+
+
+public class BibDatabaseTest {
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Before
+    public void setUp() {
+        Globals.prefs = JabRefPreferences.getInstance(); // set preferences for this test
+    }
+
+    @After
+    public void tearDown() {
+        Globals.prefs = null;
+    }
+
+    /**
+     * Some basic test cases for resolving strings.
+     *
+     * @throws FileNotFoundException
+     * @throws IOException
+     */
+    @Test
+    public void testResolveStrings() throws IOException {
+        try (FileReader fr = new FileReader("src/test/resources/net/sf/jabref/util/twente.bib")) {
+            ParserResult result = BibtexParser.parse(fr);
+
+            BibDatabase db = result.getDatabase();
+
+            assertEquals("Arvind", db.resolveForStrings("#Arvind#"));
+            assertEquals("Patterson, David", db.resolveForStrings("#Patterson#"));
+            assertEquals("Arvind and Patterson, David", db.resolveForStrings("#Arvind# and #Patterson#"));
+
+            // Strings that are not found return just the given string.
+            assertEquals("#unknown#", db.resolveForStrings("#unknown#"));
+        }
+    }
+
+    /**
+     * Tests the insertion of a {@link BibEntry} with {@link BibDatabase#insertEntry(BibEntry)}
+     */
+    @Test
+    public void insertEntry() {
+        BibDatabase db = new BibDatabase();
+        assertEquals(Collections.emptyList(), db.getEntries());
+
+        BibEntry entry = new BibEntry();
+        db.insertEntry(entry);
+        assertEquals(db.getEntries().size(), 1);
+        BibtexEntryAssert.assertEquals(entry, db.getEntries().get(0));
+    }
+
+    /**
+     * Tests {@link BibDatabase#containsEntryWithId(String)}
+     */
+    @Test
+    public void containsEntryId() {
+        BibDatabase db = new BibDatabase();
+        BibEntry entry = new BibEntry();
+        assertFalse(db.containsEntryWithId(entry.getId()));
+        db.insertEntry(entry);
+        assertTrue(db.containsEntryWithId(entry.getId()));
+    }
+
+    @Test
+    public void insertEntry_doubleValue_exceptionThrown() {
```

Comment:
use names without underscore, for example `insertEntryWithSameIdThrowsException`